core/src/main/java/org/javarosa/core/api/ClassNameHasher.java

```diff
@@ -0,0 +1,25 @@
+package org.javarosa.core.api;
+
+import org.javarosa.core.util.externalizable.Hasher;
+
+
+/**
+ * Most simple possible "hasher" - just gets the byte representation of the hash name
+ *
+ * @author wspride
+ */
+public class ClassNameHasher extends Hasher {
+
+    private final static int CLASS_HASH_SIZE = 32;
+
+    // Overrides the Hasher getHash()
+    public byte[] getHash(Class c){
+        // reverse because the beginning of the classpath is less likely to be unique than the name
+        return new StringBuilder(c.getName()).reverse().toString().getBytes();
```

Comment:
Not positive, but I'm wondering whether it's actually going to be worth caching these results. We only use this in non-performance-sensitive workflows, though, so probably not the right plan yet.